File: models/pharmacy/InsuranceClaim.js
const InsuranceClaimSchema = new mongoose.Schema({

    hospitalId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Hospital",
      required: true
    },
  
    branchId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Branch",
      required: true
    },
  
    invoiceId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "PharmacyInvoice",
      required: true
    },
  
    insuranceCompany: String,
    claimAmount: Number,
  
    claimStatus: {
      type: String,
      enum: ["Submitted","Approved","Rejected","Settled"],
      default: "Submitted"
    }
  
  }, { timestamps: true });

  

File: jstojv/src/main/java/com/samrat/jstojv/entity/InsuranceClaim.java
package com.samrat.jstojv.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "insurance_claims")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class InsuranceClaim extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "hospital_id", nullable = false)
    private Hospital hospital;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "branch_id", nullable = false)
    private Branch branch;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "invoice_id", nullable = false)
    private PharmacyInvoice invoice;

    @Column(name = "insurance_company")
    private String insuranceCompany;

    @Column(name = "claim_amount")
    private Double claimAmount;

    @Enumerated(EnumType.STRING)
    @Column(name = "claim_status")
    private Status claimStatus = Status.Submitted;

    public enum Status {
        Submitted, Approved, Rejected, Settled
    }
}
